### PR TITLE
Parameter to reduce the error to a warning #63

### PR DESCRIPTION
**Pull request by Zhack47**
Pull request for issue #63

### CODE REVIEW

File: rt_utils/rtstruct_builder.py [hunk 1]
```diff
@@ -79,7 +79,13 @@ def validate_contour_image_in_series_data(
...
-        )
+        if warning_only:
+            warnings.warn(
+                f"Loaded RTStruct references image(s) that are not contained in input series data. "
+                + f"Problematic image has SOP Instance Id: {contour_image.ReferencedSOPInstanceUID}"
+            )
+        else:
+            raise Exception(
+                f"Loaded RTStruct references image(s) that are not contained in input series data. "
+                + f"Problematic image has SOP Instance Id: {contour_image.ReferencedSOPInstanceUID}"
```
Comment (asim-shrestha) [inline rt_utils/rtstruct_builder.py:90]: Lets extract the message in a common variable and reuse within the two cases

**Review by asim-shrestha — CHANGES_REQUESTED** [review]

File: rt_utils/rtstruct_builder.py (hunk 1 above)
Comment (Zhack47) [reply rt_utils/rtstruct_builder.py]: Absolutely! Done here [41aa821](https://github.com/qurit/rt-utils/pull/64/commits/41aa8211d61f0f826fb3dcced76e0ed772269bb6)

**Review by asim-shrestha — APPROVED** [review]

File: rt_utils/rtstruct_builder.py [hunk 2]
```diff
@@ -13,14 +13,14 @@ class RTStructBuilder:
     """
 
     @staticmethod
-    def create_new(dicom_series_path: str) -> RTStruct:
+    def create_new(dicom_series_path: str, warn_only: bool = False) -> RTStruct:
```
Comment (asim-shrestha) [inline rt_utils/rtstruct_builder.py:16]: Do we need this in the new method? Looks like its only used during existing rt struct validation

File: rt_utils/rtstruct.py [hunk 3]
```diff
@@ -12,9 +12,10 @@ class RTStruct:
     Wrapper class to facilitate appending and extracting ROI's within an RTStruct
     """
 
-    def __init__(self, series_data, ds: FileDataset, ROIGenerationAlgorithm=0):
+    def __init__(self, series_data, ds: FileDataset, ROIGenerationAlgorithm=0, warn_only: bool = False):
```
Comment (asim-shrestha) [inline rt_utils/rtstruct.py:15]: Same thing as other comment, do we need this?

File: rt_utils/rtstruct_builder.py [hunk 4]
```diff
@@ -79,7 +79,9 @@ def validate_contour_image_in_series_data(
                 return
 
         # ReferencedSOPInstanceUID is NOT available
-        raise Exception(
-            f"Loaded RTStruct references image(s) that are not contained in input series data. "
-            + f"Problematic image has SOP Instance Id: {contour_image.ReferencedSOPInstanceUID}"
-        )
+        msg = f"Loaded RTStruct references image(s) that are not contained in input series data. "
```
Comment (asim-shrestha) [inline rt_utils/rtstruct_builder.py:82]: Build is failing due to indentation. Maybe replace with

```
        msg = f"Loaded RTStruct references image(s) that are not contained in input series data. " + \
            f"Problematic image has SOP Instance Id: {contour_image.ReferencedSOPInstanceUID}"
```

**Review by asim-shrestha — CHANGES_REQUESTED** [review]

File: rt_utils/rtstruct_builder.py (hunk 4 above)
Comment (Zhack47) [reply rt_utils/rtstruct_builder.py]: Done [9ca7a20](https://github.com/qurit/rt-utils/pull/64/commits/9ca7a20e00e18d36c150f2ca677756d61adeea6e)

File: rt_utils/rtstruct_builder.py (hunk 2 above)
Comment (Zhack47) [reply rt_utils/rtstruct_builder.py]: Indeed, I refactored it so it is only passed to create_new, then down to validate_rtstruct_series_references and validate_contour_image_in_series_data

File: rt_utils/rtstruct.py (hunk 3 above)
Comment (Zhack47) [reply rt_utils/rtstruct.py]: Same as above [7798fde](https://github.com/qurit/rt-utils/pull/64/commits/7798fdede0c393419ca45c133a6b0f91a6c0d290)

File: rt_utils/rtstruct_builder.py [hunk 5]
```diff
@@ -20,18 +20,18 @@ def create_new(dicom_series_path: str) -> RTStruct:
 
         series_data = image_helper.load_sorted_image_series(dicom_series_path)
         ds = ds_helper.create_rtstruct_dataset(series_data)
-        return RTStruct(series_data, ds)
+        return RTStruct(series_data, ds, warn_only=warn_only)
```
Comment (asim-shrestha) [inline rt_utils/rtstruct_builder.py:23]: Lets remove warn_only here (causing build failures)

Try running pytest in the root directory to see if failures persist

**Review by asim-shrestha — CHANGES_REQUESTED** [review]

File: rt_utils/rtstruct_builder.py (hunk 5 above)
Comment (Zhack47) [reply rt_utils/rtstruct_builder.py]: Done ,py testraaaa

**Review by asim-shrestha — APPROVED** [review]

**Comment by asim-shrestha** [comment]
Thanks for making all of the changes!

**Pull request merged**